pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -749,3 +752,9 @@ def run(self):
         deployment = deployments[0]
         deployment_path = sysroot.get_deployment_directory(deployment)
         set_system_root(deployment_path.get_path())
+
+        have_bootupd = os.path.exists(deployment_path.get_path() + '/usr/bin/bootupctl')
+        if have_bootupd:
+            log.info("Found bootupd in target root, disabling Anaconda bootloader installation")
+            bootloader = STORAGE.get_proxy(BOOTLOADER)
+            bootloader.set_use_bootupd()
```

Comment:
I don't understand the current anaconda architecture really, and I was wondering if it really works to make "cross actor" dynamic calls or not like this.  Basically here the bootloader state can only be fully computed once we've written the payload today.  Hopefully this can work.

Reply:
That won't work. The place doing `return [SomeTask()]` is called before the `run()` that checks if there is bootupd installed.

Reply:
@VladimirSlavik That's what I though as well, but the installation queue still uses workarounds from the time when the payload wasn't fully migrated, so these tasks are actually collected on demand during the installation. That makes me sad, because it means that this will work until we remove the workarounds.

On the other side, we can hook the bootloader tasks to the bootloader mode change signal and update their states on change. Alternatively, we can wrap them into another task. It won't be so difficult to support this use case.

Reply:
> Is there some way of detecting this from the payload before installation?

Not today.  we could add something to the container image metadata for it, and then have a flow that does:

- fetch metadata (i.e. container image manifest)
- set up kickstart state
- perform installation

This topic intersects with https://github.com/rhinstaller/anaconda/discussions/5197 a bit...

What I think would be really neat is generalized support for embedding kickstart fragments in the container image.  This seems most viable then if it's done in the metadata instead of the payload, because otherwise in the general case we'd need to buffer the whole payload into RAM before an install.

If we did that, then we could add e.g. `inst.bootc=quay.io/examplecorp/someos:latest` on the kernel commandline which could then entirely replace `inst.ks` - the ergonomic improvement there would be huge in the general case.

(But this general case again is for "using pre-generated os/distro ISO with custom payload", like Fedora today, but not like what we want for a custom Image Builder flow)

Reply:
Ah, actually, not having a kickstart option and state would be preferable. If there's any way to read this from the payload once it's "known", before being "installed", that would be best.

Reply:
Yep agree that's preferable!  In this stubbed out code we detect this state by noticing that `/usr/bin/bootupctl` is in the target root...that's after "install" but should still be before the bootloader stage, right?

Failing that, I think we can add a metadata property (`LABEL bootupd.enabled=true`) or so in the container image manifest.  But implementing a separate "discovery" phase would be some new code.

(Not difficult code, just need to fork off `skopeo inspect <container image reference>`)